src/ConnectionGraphicsObject.cpp

```diff
@@ -212,17 +241,39 @@ mouseReleaseEvent(QGraphicsSceneMouseEvent* event)
 
   auto node = locateNodeAt(event->scenePos(), _scene,
                            _scene.views()[0]->transform());
+  
+  if (!node) {
+    
+    if (state().requiresPort()) {
```

Comment:
This is totally broken. First, it includes a heap use after free. Second, it is an object deleting itself under disguise. It is (apart from use after free) effectively the same as:
```c++
if (state().requiresPort()) {
    _scene._temporaryConn = nullptr;
    delete this;
    return;
}
```

Is this the desired intent?

Reply:
Yeah, that's exactly what I want, and i've fixed it to be exactly the code you've written and removed the use after free.

the destructor removes all references to this object and is safe to `delete this`. I just ran it under AddressSanitizer and once the use after free was fixed, everything worked as expected. 

Thanks for the catch!